features/steps/validation_handling.py

```diff
@@ -289,6 +282,25 @@ def should_apply(items, depth):
     # evokes behave error
     generate_error_message(context, [gherkin_outcome for gherkin_outcome in context.gherkin_outcomes if gherkin_outcome.severity in [OutcomeSeverity.WARNING, OutcomeSeverity.ERROR]])
 
+
```

Comment:
I don't understand the need for breaking out `activate_rule` as a separate function.  ALB010 and SPS007 are the only two rules that use `*required*` and both do it in a `Then` step.  What am I missing?

Reply:
In gcert, we used this statement in a sequence of Given steps. For example, [here](https://github.com/buildingsmart-community/gherkin-test-cases/blob/development/features/testcases/CHILLER01M.feature). 
```
        Given an <IfcEntity>
        Given Name = <Name>
        Given A *required* relationship IfcRelAggregates to <IfcEntity> from <RelatingObject> and following that
        Given Its attribute Name

        Then Assert existence
        Then The value must be "<RelatingName>"
```
Initially, I thought we'd need to use the step this way for ALB010 as well. However, during implementation, I realized that activation wouldn't work in this case because the rule is only triggered in the `Then` step, not in the `Given` step. That's why I split the function, as it's used in two different places in the code. Later, I realized that for this rule, it would be sufficient to place the step at the end of the sequence, making it a `Then` step.

I'm unsure whether to keep the function. The advantage is that it could be very if we encounter this situation again in the future, which isn't unlikely. The disadvantage, as noticed now, is that it could be confusing. 
Should we remove it? If we decide to keep it, I'll add a note summarizing this context.

cc for @aothms for asking the same question below

Reply:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

These changes should be reverted, then we can bring the function back when we actually need it.

Reply:
Fair point, learned something new and  [reverted](https://github.com/buildingSMART/ifc-gherkin-rules/pull/284/commits/855bce36003a4e10632b3ce19f5a8c234ef4e8b5) it.